automations/js/src/project_automation/issues.mjs

```diff
@@ -7,85 +7,113 @@ import { getBoard } from '../utils/projects.mjs'
  * @param issue {import('@octokit/rest')}
  * @param board {import('../utils/projects.mjs').Project}
  * @param card {import('../utils/projects.mjs').Card}
+ * @param core {import('@actions/core')} for logging
  */
-async function syncPriority(issue, board, card) {
+async function syncPriority(issue, board, card, core) {
+  core.debug(`Starting syncPriority for issue: ${issue.number}`)
   const priority = issue.labels.find((label) =>
     label.name.includes('priority')
   )?.name
+
   if (priority) {
-    await board.setCustomChoiceField(card.id, 'Priority', priority)
+    await board.setCustomChoiceField(card.id, 'Priority', priority, core)
   }
   if (priority === '🟥 priority: critical') {
-    await board.moveCard(card.id, board.columns.ToDo)
+    await board.moveCard(card.id, board.columns.ToDo, core)
   }
 }
 
 /**
  * This is the entrypoint of the script.
  *
  * @param octokit {import('@octokit/rest').Octokit} the Octokit instance to use
+ * @param core {import('@actions/core')} GitHub Actions toolkit, for logging
  * @param context {import('@actions/github').context} info about the current event
  */
-export const main = async (octokit, context) => {
+export const main = async (octokit, core, context) => {
   const { EVENT_ACTION: eventAction } = process.env
 
+  core.debug(`Event action received: ${eventAction}`)
+
   const issue = context.payload.issue
   const label = context.payload.label
+  core.info('Issue details:', issue)
 
   if (issue.labels.some((label) => label.name === '🧭 project: thread')) {
-    // Do not add project threads to the Backlog board.
+    core.warning('Issue is a project thread. Exiting.')
     process.exit(0)
   }
 
-  const backlogBoard = await getBoard(octokit, 'Backlog')
+  await core.group('Processing Issue or PR', async () => {
+    core.debug('Getting instance for the project')
+    const backlogBoard = await getBoard(octokit, core, 'Backlog')
 
-  // Create new, or get the existing, card for the current issue.
-  const card = await backlogBoard.addCard(issue.node_id)
+    core.debug('Adding the issue or PR to the project')
+    const card = await backlogBoard.addCard(issue.node_id)
 
-  switch (eventAction) {
-    case 'opened':
-    case 'reopened': {
-      if (issue.labels.some((label) => label.name === '⛔ status: blocked')) {
-        await backlogBoard.moveCard(card.id, backlogBoard.columns.Blocked)
-      } else {
-        await backlogBoard.moveCard(card.id, backlogBoard.columns.Backlog)
-      }
-
-      await syncPriority(issue, backlogBoard, card)
-      break
-    }
+    switch (eventAction) {
+      case 'opened':
+      case 'reopened':
+        core.info('Issue opened or reopened')
```

Comment:
These comments are redundant because the switch variable `eventAction` has already been logged before.